CMakeLists.txt

```diff
@@ -14,54 +14,43 @@ find_package(Eigen3 REQUIRED)
 
 include_directories(
   ${PROJECT_SOURCE_DIR}/include
-  ${PROJECT_SOURCE_DIR}/third_party
+  ${PROJECT_SOURCE_DIR}/third_party/Catch2/single_include/catch2
   ${EIGEN3_INCLUDE_DIR}
   ${QPOASES_INCLUDE_DIR}
 
 )
 
 
 add_executable(
-  catchbezier
-  test/catch_bezier.cpp
+  beziertests
+  test/bezier_tests.cpp
   test/main.cpp
 )
 
-target_link_libraries(
-catchbezier
-qpOASES
-)
-
 add_executable(
-bsplinetest
-test/testbspline.cpp
+bsplinetests
+test/bspline_tests.cpp
+test/main.cpp
 )
 
-target_link_libraries(
-bsplinetest
-qpOASES
-)
 
 add_executable(
-beziertest
-test/testbezier.cpp
+curvetests
+test/curve_tests.cpp
+test/main.cpp
 )
 
-target_link_libraries(
-beziertest
-qpOASES
-)
 
 add_executable(
-splinetest
-test/testspline.cpp
+splinetests
+test/spline_tests.cpp
+test/main.cpp
 )
 
-target_link_libraries(
-splinetest
-qpOASES
-)
 
 
 include(CTest)
-add_test(NAME RunTests COMMAND $<TARGET_FILE:catchbezier>)
```

Comment:
instead of adding tests directly to the main cmake file what do you think about doing something like this:

add a CMakeLists file to tests directory. include it to the main cmakelists(by add subdirectory) if BUILD_TESTING is enabled. I think this would seperate two things better.

Check these:
https://cmake.org/cmake/help/latest/module/CTest.html
https://stackoverflow.com/a/50476125

Reply:
are we planning to build anything other than tests in this repository? If that's the case, it would be better to separate as you said

Reply:
well, not currently. but we may in the future especially if we are going to generate docs automatically. I guess separating it is better anyway.

Reply:
Okay, I will be sending a pull request soon